addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -900,7 +900,7 @@ var FieldX2Many = AbstractField.extend({
             // Correctly resets the cursor
             var self = this;
             var editableId =  this.renderer.getEditableRecordID();
-            var datapoint = _.find(this.value.data, {id: editableId});
```

Comment:
the fact that the value is different is the problem. Do you know why they get desynchronized at some point?